geomstats/geometry/grassmannian.py

```diff
@@ -50,9 +50,33 @@ def belongs(self, point, tolerance=TOLERANCE):
         -------
         belongs : bool
         """
-        raise NotImplementedError(
-            'The Grassmann `belongs` is not implemented.'
-            'It shall test whether p*=p, p^2 = p and rank(p) = k.')
+        point = gs.to_ndarray(point, to_ndim=3)
+        n_points, n, p = point.shape
+
+        # check square
+        if (n, p) != (self.n, self.n):
+            return gs.array([[False]] * n_points)
+
+        point_transpose = gs.transpose(point, axes=(0, 2, 1))
```

Comment:
You can use Matrices.transpose(point) here. (no need to specify the axes this way, and it will help broadcasting).